tree/ntuple/v7/src/RField.cxx

```diff
@@ -1687,6 +1721,114 @@ void ROOT::Experimental::RClassField::AcceptVisitor(Detail::RFieldVisitor &visit
 
 //------------------------------------------------------------------------------
 
+ROOT::Experimental::RUnsplitField::RUnsplitField(std::string_view fieldName, std::string_view className,
+                                                 std::string_view typeAlias)
+   : RUnsplitField(fieldName, className, TClass::GetClass(std::string(className).c_str()))
+{
+   fTypeAlias = typeAlias;
+}
+
+ROOT::Experimental::RUnsplitField::RUnsplitField(std::string_view fieldName, std::string_view className, TClass *classp)
+   : ROOT::Experimental::RFieldBase(fieldName, className, ENTupleStructure::kUnsplit, false /* isSimple */),
+     fClass(classp),
+     fIndex(0)
+{
+   if (fClass == nullptr) {
+      throw RException(R__FAIL("RUnsplitField: no I/O support for type " + std::string(className)));
+   }
+
+   if (!(fClass->ClassProperty() & kClassHasExplicitCtor))
+      fTraits |= kTraitTriviallyConstructible;
+   if (!(fClass->ClassProperty() & kClassHasExplicitDtor))
+      fTraits |= kTraitTriviallyDestructible;
+}
+
+std::unique_ptr<ROOT::Experimental::RFieldBase>
+ROOT::Experimental::RUnsplitField::CloneImpl(std::string_view newName) const
+{
+   return std::unique_ptr<RUnsplitField>(new RUnsplitField(newName, GetTypeName(), fClass));
+}
+
+std::size_t ROOT::Experimental::RUnsplitField::AppendImpl(const void *from)
+{
+   TBufferFile buffer(TBuffer::kWrite, GetValueSize());
+   fClass->Streamer(const_cast<void *>(from), buffer);
+
+   auto nbytes = buffer.Length();
+   fColumns[1]->AppendV(buffer.Buffer(), buffer.Length());
+   fIndex += nbytes;
+   fColumns[0]->Append(&fIndex);
+   return nbytes + fColumns[0]->GetElement()->GetPackedSize();
+}
+
+void ROOT::Experimental::RUnsplitField::ReadGlobalImpl(NTupleSize_t globalIndex, void *to)
+{
+   RClusterIndex collectionStart;
+   ClusterSize_t nbytes;
+   fPrincipalColumn->GetCollectionInfo(globalIndex, &collectionStart, &nbytes);
+
+   TBufferFile buffer(TBuffer::kRead, nbytes);
+   fColumns[1]->ReadV(collectionStart, nbytes, buffer.Buffer());
```

Comment:
(Especially since the buffer is transient), we might (or might not) be able to use the `TBuffer`'s mechanism to use a non-owned memory area to avoid one memcpy.

Reply:
I think it will be difficult because the buffer is constructed from potentially several pages.

Reply:
fair enough.